lib/ts/recipe/thirdParty/coreAPICalls.ts

```diff
@@ -0,0 +1,104 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Recipe from "./recipe";
+import NormalisedURLPath from "../../normalisedURLPath";
+import { User } from "./types";
+
+export async function signInUp(
+    recipeInstance: Recipe,
+    thirdPartyId: string,
+    thirdPartyUserId: string,
+    email: {
+        id: string;
+        isVerified: boolean;
+    }
+): Promise<{ isNewUser: boolean; user: User } | undefined> {
```

Comment:
- Why are you returning undefined
- Why have you changed `createdNewUser` to `isNewUser`?